### PR TITLE
Multiple Tile Sets Based On Zoom Level

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Features "Multiple Tile Sets Based On Zoom Level"

#### Purpose of change
Sometimes I strain my eyes staring at ultica when it is far away. This gives the fidelity of ultica with another option

#### Describe the solution
Game can now load a second tileset just for far away. Smoothly swaps between them as you zoom in and out.

Store the tilesets seperately and as you zoom it swaps the tileset that the game uses between them (with pointers).

Added relevant options as well including custom setting the threshold where this happens

#### Testing
Game loads and should do it fine.

#### Additional context
image

https://user-images.githubusercontent.com/4514073/181286464-208e2045-88a4-416b-adc5-0ec0eae5c1d7.mp4

### CODE REVIEW

**Comment by Zireael07** [comment]
As a glasses user, THANK YOU!

**Comment by Qrox** [comment]
Will this increase memory consumption even if the far-away tileset is disabled (or set to the same as the close-up tileset)? Seems to be the case because the far-away tileset is unconditionally loaded, unless there's some caching around there that I'm unaware of. I also feel like a better option is too allow tilesets to provide smaller-size sprites for far-away rendering, but that's way out of the scope of this PR I suppose.

**Comment by bombasticSlacks** [comment]
Its reasonable that memory footprint is a concern I can easily disable it and check if the close and far are identical.

File: src/options.cpp [hunk 1]
```diff
@@ -2058,7 +2058,19 @@ void options_manager::add_options_graphics()
          build_tilesets_list(), "UltimateCataclysm", COPT_CURSES_HIDE
        ); // populate the options dynamically
 
+    add( "USE_DISTANT_TILES", "graphics", to_translation( "Use seperate tileset for far" ),
```
Comment (Night-Pryanik) [inline src/options.cpp:2061]:
```suggestion
    add( "USE_DISTANT_TILES", "graphics", to_translation( "Use separate tileset for far" ),
```

File: src/options.cpp [hunk 2]
```diff
@@ -2058,7 +2058,19 @@ void options_manager::add_options_graphics()
...
+         to_translation( "If true, when very zoomed out you will use a seperate tileset." ),
```
Comment (Night-Pryanik) [inline src/options.cpp:2062]:
```suggestion
         to_translation( "If true, when very zoomed out you will use a separate tileset." ),
```

**Comment by ANickelN** [comment]
Amazing change, going to love trying this out in-game.

**Pull request merged**